casbin/fast_enforcer.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from typing import Sequence
+
+from casbin.enforcer import Enforcer
+from casbin.model import Model, FastModel, fast_policy_filter, FunctionMap
+from casbin.persist.adapters import FileAdapter
+from casbin.util.log import configure_logging
+
+
+class FastEnforcer(Enforcer):
+    _cache_key_order: Sequence[int] = None
+
+    def __init__(self, model=None, adapter=None, enable_log=False, cache_key_order: Sequence[int] = None):
+        self._cache_key_order = cache_key_order
+        self.logger = logging.getLogger("casbin.enforcer")
+        if isinstance(model, str):
+            if isinstance(adapter, str):
+                self.init_with_file(model, adapter)
+            else:
+                self.init_with_adapter(model, adapter)
+                pass
+        else:
+            if isinstance(adapter, str):
+                raise RuntimeError("Invalid parameters for enforcer.")
+            else:
+                self.init_with_model_and_adapter(model, adapter)
+
+        if enable_log:
+            configure_logging()
+
+    def init_with_file(self, model_path, policy_path):
+        """initializes an enforcer with a model file and a policy file."""
+        a = FileAdapter(policy_path)
+        self.init_with_adapter(model_path, a)
+
+    def init_with_adapter(self, model_path, adapter=None):
+        """initializes an enforcer with a database adapter."""
+        m = self.new_model(model_path)
+        self.init_with_model_and_adapter(m, adapter)
+
+        self.model_path = model_path
```

Comment:
There is a lot of duplicate code here and `core_enforcer`. Just use `super().__init__()` in init to initialize the parent class